### PR TITLE
Add method to find possible meeting times.

### PR DESCRIPTION
**Pull request by juanuribe28**

### CODE REVIEW

File: walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java [hunk 1]
```diff
@@ -22,6 +22,11 @@
    * Returns a greeting for the given name.
    */
   public String greet(String name) {
+    name = removeWhitespaces(name);
     return "Hello " + name;
   }
+
+  public String removeWhitespaces(String string) {
```
Comment (seangilpin) [inline walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:29]: If the string is "Sean Gilpin" this should produce "SeanGilpin" correct? Just want to confirm thats what you want.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,49 @@
...
+        continue;
+      }
+      TimeRange eventTime = event.getWhen();
+      for (TimeRange meetingTime : meetingTimes) {
+        if (eventTime.overlaps(meetingTime)) {
+          meetingTimes.remove(meetingTime);
+          if (meetingTime.start() < eventTime.start()) {
+            TimeRange before = TimeRange.fromStartEnd(meetingTime.start(), eventTime.start(), false);
+            if (before.duration() >= request.getDuration()) {
+              meetingTimes.add(before);
```
Comment (seangilpin) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:47]: Is adding to meetingTimes while looping through it a supported operation?

https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html

Comment (juanuribe28) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:47]: Yeah, I already tested the code and it works.

File: walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java (hunk 1 above)
Comment (juanuribe28) [reply walkthroughs/week-5-tdd/intro/src/main/java/com/google/sps/Greeter.java:29]: That's correct.

File: walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java [hunk 3]
```diff
@@ -30,4 +30,14 @@ public void testGreeting() {
...
     Assert.assertEquals("Hello Ada", greeting);
   }
+
+  @Test
+  public void testGreetingTrimsWhitespace() {
+    Greeter greeter = new Greeter();
+
+    String greeting = greeter.greet("   Ada   ");
+
+    // Whitespace should be trimmed
```
Comment (daperlmut) [inline walkthroughs/week-5-tdd/intro/src/test/java/com/google/sps/GreeterTest.java:40]: [Nit] Add a '.'

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 4]
```diff
@@ -15,9 +15,49 @@
...
+            }
+          }
+          if (eventTime.end() < meetingTime.end()) {
+            TimeRange after = TimeRange.fromStartEnd(eventTime.end(), meetingTime.end(), false);
+            if (after.duration() >= request.getDuration()) {
+              meetingTimes.add(after);
+            }
+          }
+        }
```
Comment (daperlmut) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: This is very deeply nested code which should generally be avoid as it makes it somewhat hard to reason about.  For example, what is the set of all conditions that must be true to reach line 51?

My suggestion is to factor out the inside of the for loop into a function and use early returns to make the preconditions a bit more clear.

Comment (juanuribe28) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: Should I do this change in this PR, or should I open and Issue for a new one?

Comment (daperlmut) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: Hmm, I think either is fine.  Feel free to add a TODO and change it in the next PR if you want.

Comment (juanuribe28) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:56]: Ok, I opened a new issue about it.

**Pull request merged**